lsp-semantic-tokens.el

```diff
@@ -541,6 +571,149 @@ LOUDLY will be forwarded to OLD-FONTIFY-REGION as-is."
                when (> current-line line-max-inclusive) return nil)))))
       `(jit-lock-bounds ,beg . ,end)))))
 
+(defun lsp-semantic-tokens-toggle-comment-syntax ()
+  "Toggle the value of `lsp-semantic-tokens-set-comment-syntax'"
+  (interactive)
+  (if lsp-semantic-tokens-set-comment-syntax
+      (progn
+        (setq lsp-semantic-tokens-set-comment-syntax nil)
+        (lsp-semantic-tokens--remove-comment-syntax (point-min) (point-max)))
+    (setq lsp-semantic-tokens-set-comment-syntax t))
+  (font-lock-fontify-buffer))
+
+(defun lsp-semantic-tokens--get-overlapping-comments (beg end)
+  "Returns the start of the comment pair that contains BEG, and the
+end of the comment pair that contains END. If either of such
+comment pairs does not exist, return nil for that part. This
+function also removes dangling comment starters/ends."
+  (let (prev-beg
+        prev-end
+        next-beg
+        next-end
+        tmp)
+    (save-mark-and-excursion
```

Comment:
nit: use `lsp-save-restriction-and-excursion` (here and on a few more places)

Reply:
Done.